### PR TITLE
Fix databricks experience

### PR DESCRIPTION
**Pull request by xiaoyongzhu**
In this PR we have added a few fixes:
- when the input databricks config is not a string but a dict, it can be parsed
- result of databricks can be downloaded (if it's in dbfs)

### CODE REVIEW

File: feathr_project/feathr/_databricks_submission.py [hunk 1]
```diff
@@ -192,4 +198,43 @@ def get_job_result_uri(self) -> str:
...
+            for file_path in dbfs_files:
+                # each file_path would be a dict of this type: https://docs.microsoft.com/en-us/azure/databricks/dev-tools/api/latest/dbfs#dbfsfileinfo
+                dbfs_file_path, dbfs_file_size, local_file_path = file_path['path'], file_path['file_size'], os.path.join(local_folder, os.path.basename(file_path['path']))
+                with open(local_file_path, 'wb') as file_obj:
+                    downloaded_size = 0
+                    # Loop until we've downloaded the whole file
+                    while downloaded_size < dbfs_file_size:
+                        chunk = self._read_single_chunk(path=dbfs_file_path, offset=downloaded_size, length=DATABRICKS_MB_BYTES)
+                        file_obj.write(base64.b64decode(chunk.data))
+                        downloaded_size += chunk.bytes_read
```
Comment (hangfei) [inline feathr_project/feathr/_databricks_submission.py:222]: what if the file is too large? Can we notify how much data are we going to download and notify the user the progres?

File: feathr_project/feathr/_databricks_submission.py [hunk 2]
```diff
@@ -192,4 +198,43 @@ def get_job_result_uri(self) -> str:
...
+
+    def _read_single_chunk(self, path, offset, length=DATABRICKS_MB_BYTES):
+
+        params = {"path": path,
+                "offset": offset,
+                "length": length}
+
+        # get a single chunk of file
+        # https://docs.microsoft.com/en-us/azure/databricks/dev-tools/api/latest/dbfs#--read
+        resp = requests.get(url=self.workspace_instance_url+'/api/2.0/dbfs/read',  headers=self.auth_headers,  params=params)
```
Comment (hangfei) [inline feathr_project/feathr/_databricks_submission.py:235]: let's not duplicate the code.

Comment (xiaoyongzhu) [reply feathr_project/feathr/_databricks_submission.py:235]: It's not a duplicated code. This function is to read single chunk of file (up to 1MB) per Databricks API.

File: feathr_project/feathr/_databricks_submission.py [hunk 3]
```diff
@@ -192,4 +200,49 @@ def get_job_result_uri(self) -> str:
                                headers=self.auth_headers, params={'run_id': str(self.res_job_id)})
         custom_tags = result.json()['cluster_spec']['new_cluster']['custom_tags']
         assert custom_tags is not None
-        return custom_tags[OUTPUT_PATH_TAG]
\ No newline at end of file
+        return custom_tags[OUTPUT_PATH_TAG]
+
+    def download_result(self, result_path: str, local_folder: str):
```
Comment (hangfei) [inline feathr_project/feathr/_databricks_submission.py:205]: add a test for this?

File: feathr_project/feathr/_databricks_submission.py [hunk 4]
```diff
@@ -192,4 +200,49 @@ def get_job_result_uri(self) -> str:
...
+        """
+        Supports downloading files from the result folder. Only support paths starts with `dbfs:/` and only support downloading files in one folder (per Spark's design, everything will be in the result folder in a flat manner)
+        """
+        print(result_path,result_path.startswith('dfbs'))
+        if not result_path.startswith('dfbs'):
+            RuntimeError('Currently only paths starting with dbfs is supported for downloading results from a databricks cluster. The path should start with \"dbfs:\" .')
```
Comment (hangfei) [inline feathr_project/feathr/_databricks_submission.py:211]: add a test for this?

**Review by hangfei — APPROVED** [review]

**Review by jaymo001 — APPROVED** [review]

**Pull request merged**